plugins/org.komodo.core/src/org/komodo/repository/RepositoryImpl.java

```diff
@@ -1243,4 +1254,18 @@ public void unpublish( final UnitOfWork uow,
 
     }
 
+    @Override
+    public KomodoObject komodoRoot(final UnitOfWork uow) throws KException {
+        return create(uow, KOMODO_ROOT, KomodoLexicon.Komodo.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoLibrary(final UnitOfWork uow) throws KException {
+        return create(uow, LIBRARY_ROOT, KomodoLexicon.Library.NODE_TYPE);
+    }
+
+    @Override
+    public KomodoObject komodoWorkspace(final UnitOfWork uow) throws KException {
```

Comment:
Was thinking about this whilst out ... Can't we remove the `komodoRoot` method from the interface? Just have it a private method in the impl. It doesn't seem like any caller of the repository would ever need the root directly.